core/chain_manager.go

```diff
@@ -48,6 +48,9 @@ var (
 )
 
 const (
+	headerCacheLimit    = 512
+	bodyCacheLimit      = 256
+	tdCacheLimit        = 1024
```

Comment:
Since the TD has its own accessor functions now, I imagine you could find other uses for it without a header too (the downloader regularly retrieves the head TD). I think really the reason was space consumption. Even if the TDs are always linked to the headers, reaching out fewer times to the database could be a good thing. Given that a TD is probably 4-6 bytes, I don't think it matters much how many we store. Note, the numbers here are mostly arbitrary, they were introduced in Olympic to reduce the database trashing a bit, but they should eventually be fine tuned. Unfortunately for that you would need much heavier usage and some realistic way to benchmark them. For now I would try to go with sane default and then maybe keep an eye on them if some issues crop up.